api/client.go

```diff
@@ -98,3 +98,33 @@ func (c *Client) Pull(ctx context.Context, req *PullRequest, fn PullProgressFunc
 		return fn(resp)
 	})
 }
+
+func (c *Client) Embedding(ctx context.Context, req EmbeddingRequest) (*EmbeddingResponse, error) {
```

Comment:
This is a self-contained client function because I believe Mike mentioned something about refactoring the client. I wanted to avoid overlapping work there.